cli/src/test/java/com/devonfw/tools/ide/context/IdeContextTest.java

```diff
@@ -128,4 +130,43 @@ public void testCurrentWorkspacePathIsUsed() {
     assertThat(workspaceName).isEqualTo("foo-test");
   }
 
+  // hier test einfügen mit idetestcontext
+  @Test
+  public void testIdeVersionTooSmall() {
+    // arrange
+    String path = "project/workspaces/foo-test";
+    IdeTestContext context = newContext(PROJECT_BASIC, path, false);
+    EnvironmentVariables variables = context.getVariables();
+    String ideMinVersion = String.valueOf(Integer.MAX_VALUE);
+    variables.getByType(EnvironmentVariablesType.CONF).set("IDE_MIN_VERSION", ideMinVersion);
+    CliArguments args = new CliArguments();
+    String warningMessage = String.format("Your version of IDEasy is currently %s\n"
+        + "However, this is too old as your project requires at latest version %s\n"
+        + "Please run the following command to update to the latest version of IDEasy and fix the problem:\n"
+        + "ide upgrade", IdeVersion.getVersionIdentifier().toString(), ideMinVersion);
+    // act
+    context.run(args);
+    // assert
+    assertThat(context).logAtWarning().hasMessage(warningMessage);
+  }
+
+  @Test
+  public void testIdeVersionOk() {
+    // arrange
+    String path = "project/workspaces/foo-test";
+    IdeTestContext context = newContext(PROJECT_BASIC, path, false);
+    EnvironmentVariables variables = context.getVariables();
+    String ideVersion = IdeVersion.getVersionIdentifier().toString();
+    variables.getByType(EnvironmentVariablesType.CONF).set("IDE_MIN_VERSION", ideVersion);
+    CliArguments args = new CliArguments();
+    String warningMessage = String.format("Your version of IDEasy is currently %s\n"
+        + "However, this is too old as your project requires at latest version %s\n"
+        + "Please run the following command to update to the latest version of IDEasy and fix the problem:\n"
+        + "ide upgrade", ideVersion, ideVersion);
+    // act
+    context.run(args);
+    // assert
+    assertThat(context).logAtWarning().hasNoMessage(warningMessage);
```

Comment:
just a nice to have:
But there is also `hasNoMessageContaining` so you could just search for a substring like `is too old as your project requires at least`. Also you might want to replace `logAtWarning` with just `log` here since we do not expect this message regardless of the log-level.